lago/sdk_utils.py

```diff
@@ -0,0 +1,74 @@
+import functools
+import inspect
+import wrapt
+
+
+class SDKWrapper(wrapt.ObjectProxy):
+    """A proxy object that exposes only methods which were decorated with
+    :func:`expose` decorator."""
+
+    def __getattr__(self, name):
+        attr = getattr(self.__wrapped__, name)
+        return filter_sdk(attr, name)
+
+    def __dir__(self):
+        orig = super(SDKWrapper, self).__dir__()
+        filtered = []
+        for name in orig:
+            attr = getattr(self.__wrapped__, name)
+            try:
+                filter_sdk(attr, name)
+                filtered.append(name)
+            except AttributeError:
+                pass
+        return filtered
+
+
+class SDKMethod(object):
+    """Metadata to store inside the decorated function"""
+
+    def __init__(self, name):
+        self.name = name
+
+
+def expose(func):
+    """
+    Decorator to be used with :class:`SDKWrapper`. This decorator indicates
+    that the wrapped method or class should be exposed in the proxied object.
+
+    Args:
+        func(types.FunctionType/types.MethodType): function to decorate
+
+    Returns:
+        None
+    """
+
+    @functools.wraps(func)
+    def wrapped(*args, **kwargs):
+        return func(*args, **kwargs)
+
+    if inspect.isclass(func):
+        wrapped._sdkmetaclass = SDKMethod(func.__name__)
+    else:
+        wrapped._sdkmeta = SDKMethod(func.__name__)
+    return wrapped
+
+
+def filter_sdk(attr, name):
+    """
+    Filter SDK attributes
+
+    Args:
+        attr(attribute): Attribute as returned by :func:`getattr`.
+        name(str): Attribute name.
+
+    Returns:
+        `attr` if passed.
+    """
+
+    if name.startswith('_'):
+        return attr
```

Comment:
OK - removed the entire detection in https://github.com/lago-project/lago/pull/531/commits/4d184a839a7cce651ac9d1fb50e109a855f67850
Will only add "expose" methods.